### PR TITLE
sysinfo and podrescompat removal: superseded by getallocatableResources

### PR DESCRIPTION
**Pull request by swatisehgal**
`sysinfo` and `podrescompat` packages were introduced as a way to obtain resource information from the system and serve the endpoints provided by podresources in the absence of getAllotableResource endpoint in podresource API.

Support for getallocatableResource endpoint graduated to GA recently and it is safe to assume that this endpoint is enabled by default and provides all the necesary information for determining node resource allocatable.

We therefore no longer need the aforementioned packages and remove them.

### CODE REVIEW

**Review by ffromani — COMMENTED** [review]
nice work, thanks. Some comments inline

**Comment by ffromani** [comment]
I think we should also do a pass of `go mod tidy && go mod vendor` in the context of this PR (if you resubmit AND CI doesn't complain - it seems it doesn't)

File: rte/main.go [hunk 1]
```diff
@@ -86,38 +82,18 @@ func main() {
...
-	klog.Infof(`
-=== System information ===
-%s
-==========================`, sysInfo)
-
-	if parsedArgs.SysinfoOnly {
-		os.Exit(0)
-	}
-
 	k8sCli, err := podres.GetClient(parsedArgs.RTE.PodResourcesSocketPath)
```
[the remark this answers is not in the document]
Comment (swatisehgal) [reply rte/main.go:85]: Ack.

File: rte/main.go [hunk 2]
```diff
@@ -86,30 +82,13 @@ func main() {
...
 		klog.Fatalf("failed to start prometheus server: %v", err)
 	}
 
-	sysCli := k8sCli
-	if !parsedArgs.LocalArgs.SysConf.IsEmpty() {
-		sysCli = podrescompat.NewSysinfoClientFromLister(k8sCli, parsedArgs.LocalArgs.SysConf)
-	}
-
-	cli := sharedcpuspool.NewFromLister(sysCli, parsedArgs.RTE.Debug, parsedArgs.RTE.ReferenceContainer)
+	sysCli := cli
```
Comment (ffromani) [inline rte/main.go:90]: no need, we can just use
```
cli = sharedcpuspool.NewFromLister(sysCli, parsedArgs.RTE.Debug, parsedArgs.RTE.ReferenceContainer)
```

**Review by ffromani — APPROVED** [review]
LGTM

minor nit, please don't bother unless you need to resubmit for other reasons

File: rte/main.go (hunk 2 above)
Comment (swatisehgal) [reply rte/main.go:90]: Sure, can fix this as well.

**Pull request merged**